public/app/core/logs_model.ts

```diff
@@ -198,16 +197,33 @@ function isLogsData(series: DataFrame) {
 export function dataFrameToLogsModel(
   dataFrame: DataFrame[],
   intervalMs: number | undefined,
-  timeZone: TimeZone
+  timeZone: TimeZone,
+  absoluteRange?: AbsoluteTimeRange
 ): LogsModel {
   const { logSeries, metricSeries } = separateLogsAndMetrics(dataFrame);
   const logsModel = logSeriesToLogsModel(logSeries);
 
   if (logsModel) {
     if (metricSeries.length === 0) {
-      // Create metrics from logs
-      // If interval is not defined or 0 we cannot really compute the series
-      logsModel.series = intervalMs ? makeSeriesForLogs(logsModel.rows, intervalMs, timeZone) : [];
+      // Create histogram metrics from logs using the interval as bucket size for the line count
+      if (intervalMs && logsModel.rows.length > 0) {
+        const sortedRows = logsModel.rows.sort(sortInAscendingOrder);
+        let resolutionIntervalMs = intervalMs;
+        // Clamp time range to visible logs otherwise big parts of the graph might look empty
+        if (absoluteRange) {
+          const earliest = sortedRows[0].timeEpochMs;
```

Comment:
```suggestion
          const earliest = sortedRows[0].timeEpochMs - 30000;
```
I think it would make sense to shift this a little because if it is too close to the earliest value then it can not bee seen in the graph.
![screenshot-localhost_3000-2020 05 25-14_02_58](https://user-images.githubusercontent.com/13729989/82811400-f28c9480-9e90-11ea-9c18-c16b7637e9d7.png)
after subtracting 30 seconds from the earliest time: 
![screenshot-localhost_3000-2020 05 25-14_04_12](https://user-images.githubusercontent.com/13729989/82811412-fa4c3900-9e90-11ea-89f7-96067d1b1694.png)

Reply:
I would guess it's an alignment issue and it should be `sortedRows[0].timeEpochMs - intervalMs` or something like that. We may need to pull the interval calculation so it can be reused or moved here.

Reply:
Yeah, the timeseries generation does a flooring of bucketed times:
```
const time = Math.floor(row.timeEpochMs / bucketSize) * bucketSize;
```
So the visible range must do a similar thing otherwise the first datapoint is not rendered. Great catch @zoltanbedi